tiledb/sm/group/group.h

```diff
@@ -425,6 +430,9 @@ class Group {
   /** Mutex for thread safety. */
   mutable std::mutex mtx_;
 
+  /** The ContextResources class. */
+  std::reference_wrapper<ContextResources> resources_;
```

Comment:
What's the reason for `reference_wrapper` here? It looks like we don't need it. The constructor takes a `ContextResources&` argument and initializes the member variable. This class represents a resource, so we don't want it to be copy-assignable (which should be deleted, but isn't), nor copy-constructible (ditto).